### PR TITLE
plot: Fill Between

### PR DESCRIPTION
**Pull request by johannes-wolf**
This adds a new `plot.add-fill-between(a, b, ..)` to fill the area between two line charts,
useful for plotting an error-band along a function.

### CODE REVIEW

File: src/lib/plot/line.typ [hunk 1]
```diff
@@ -317,3 +317,103 @@
...
+
+/// Fill the area between two graphs. This behaves same as `plot` but takes
+/// a pair of data instead of a single data array/function.
+/// The area between both function plots gets filled.
+///
+/// This can be used to display an error-band of a function.
+///
+/// - domain (array): Domain tuple of the plot. If `data-a` or `data-b` is a function,
+///                   domain must be specified, as `data-a` or `data-b` is sampled
+///                   for x-values in `domain`. Values must be numbers.
```
Comment (fenjalien) [inline src/lib/plot/line.typ:329]: If domain is used in every function for the plot lib would it be worth defining a domain type somewhere?

File: src/lib/plot/line.typ [hunk 2]
```diff
@@ -317,3 +317,103 @@
...
+/// - samples (int): Number of times the `data-a` and `data-b` function gets called for
+///                  sampling y-values. Only used if `data-a` or `data-b` is of
+///                  type function.
+/// - sample-at (array): Array of x-values the function(s) get sampled at in addition
+///                      to the default sampling.
+/// - line (string, dictionary): Line type to use, see `add`
+/// - style (style): Style to use, can be used with a palette function
+/// - axes (array): Name of the axes to use ("x", "y"), note that not all
+///                 plot styles are able to display a custom axis!
```
Comment (fenjalien) [inline src/lib/plot/line.typ:338]: Names of axes to use for what?

**Comment by fenjalien** [comment]
Forgot to add: the manual needs updating :)

File: src/lib/plot/line.typ (hunk 1 above)
Comment (johannes-wolf) [reply src/lib/plot/line.typ:329]: Yes, same for axes. :+1:

**Comment by johannes-wolf** [comment]
Updated.

**Review by fenjalien — APPROVED** [review]

**Pull request merged**